src/platform/plugins/shared/workflows_management/public/widgets/workflow_yaml_editor/lib/snippets/insert_trigger_snippet.ts

```diff
@@ -7,47 +7,131 @@
  * License v3.0 only", or the "Server Side Public License, v 1".
  */
 
-import { type Document } from 'yaml';
+import { type Document, parseDocument, type Range } from 'yaml';
+import { isSeq } from 'yaml';
 import { monaco } from '@kbn/monaco';
 import type { TriggerType } from '@kbn/workflows';
 import { generateTriggerSnippet } from './generate_trigger_snippet';
-import { getTriggerNodes } from '../../../../../common/lib/yaml';
+import { getTriggerNodes, getTriggersPair } from '../../../../../common/lib/yaml';
 import { getIndentLevelFromLineNumber } from '../get_indent_level';
 import { prependIndentToLines } from '../prepend_indent_to_lines';
-import { getMonacoRangeFromYamlNode } from '../utils';
+import { getMonacoRangeFromYamlNode, getMonacoRangeFromYamlRange } from '../utils';
 
-// Algorithm:
-// 1. Find the next line after the last trigger node range
-// 2. If no trigger nodes found, add triggers: section in the first line of yaml
+/**
+ * Gets the triggers key range and calculates the expected indent level for array items
+ */
+function getTriggersKeyInfo(
+  model: monaco.editor.ITextModel,
+  triggersPair: ReturnType<typeof getTriggersPair>
+): { range: monaco.Range | null; indentLevel: number } {
+  if (!triggersPair?.key) {
+    return { range: null, indentLevel: 2 };
+  }
+
+  const triggersKey = triggersPair.key;
+  const triggersKeyRange =
+    typeof triggersKey === 'object' &&
+    triggersKey !== null &&
+    'range' in triggersKey &&
+    triggersKey.range
+      ? getMonacoRangeFromYamlRange(model, triggersKey.range as Range)
+      : null;
+
+  const indentLevel = triggersKeyRange
+    ? getIndentLevelFromLineNumber(model, triggersKeyRange.startLineNumber) + 2
+    : 2;
+
+  return { range: triggersKeyRange, indentLevel };
+}
+
+/**
+ * Creates a replacement range for an empty item line
+ */
+function createReplacementRange(model: monaco.editor.ITextModel, lineNumber: number): monaco.Range {
+  const nextLineNumber = lineNumber + 1;
+  if (nextLineNumber <= model.getLineCount()) {
+    return new monaco.Range(lineNumber, 1, nextLineNumber, 1);
+  }
+  const lineEndColumn = model.getLineMaxColumn(lineNumber);
+  return new monaco.Range(lineNumber, 1, lineNumber, lineEndColumn);
+}
 
+// Algorithm:
+// 1. Check if triggers section exists (even if empty or has empty items)
+// 2. If triggers section exists, find the next line after the last trigger node range
+// 3. If no trigger section found, add triggers: section in the first line of yaml
 export function insertTriggerSnippet(
   model: monaco.editor.ITextModel,
-  yamlDocument: Document,
+  yamlDocument: Document | null,
   triggerType: TriggerType,
   editor?: monaco.editor.IStandaloneCodeEditor
 ) {
-  // find triggers: line number and column number
-  const triggerNodes = getTriggerNodes(yamlDocument);
+  const document = yamlDocument || parseDocument(model.getValue());
```

Comment:
consider catching potential exceptions from parseDocument (doc can be malformed)